optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,733 @@
+            vision_config_dict = (
+                model_cfg["vision_cfg"] if model_cfg.get("vision_cfg", None) else model_cfg["vision_config"]
+            )
+
+            config = CLIPConfig(
+                text_config_dict=text_config_dict,
+                vision_config_dict=vision_config_dict,
+                **open_clip_config,
+            )
+        except OSError as e:
+            # if config not found in subfolder, search for it at the top level
+            if subfolder != "":
+                config = AutoConfig.from_pretrained(
+                    pretrained_model_name_or_path=config_name_or_path,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    token=token,
+                    trust_remote_code=trust_remote_code,
+                )
+                logger.info(
+                    f"config.json not found in the specified subfolder {subfolder}. Using the top level config.json."
+                )
+            else:
+                raise OSError(e)
+        return config
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        export: bool = False,
+        config: Optional["PretrainedConfig"] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        if isinstance(model_id, Path):
+            model_id = model_id.as_posix()
+
+        from_transformers = kwargs.pop("from_transformers", None)
+        if from_transformers is not None:
+            logger.warning(
+                "The argument `from_transformers` is deprecated, and will be removed in optimum 2.0.  Use `export` instead"
+            )
+            export = from_transformers
+
+        if len(model_id.split("@")) == 2:
+            if revision is not None:
+                logger.warning(
+                    f"The argument `revision` was set to {revision} but will be ignored for {model_id.split('@')[1]}"
+                )
+            model_id, revision = model_id.split("@")
+
+        config_path = config if isinstance(config, (str, os.PathLike)) else model_id
```

Comment:
if this is the only difference then I think we don't need to rewrite `from_pretrained` but only load the config before calling super().from_pretrained(..) wdyt ?

Reply:
OptimizedModel.from_pretrained use also `TasksManager.infer_library_from_model` , which will fails for open_clip

Reply:
ok so this should be added in optimum but let's keep it in optimum-intel for now, can you add a comment to specify that this should be removed when added to optimum (when compatible with TasksManager.infer_library_from_model)

Reply:
thanks!